src/trajectory.jl

```diff
@@ -223,16 +221,26 @@ end
 
 ### Use end-point from trajecory as proposal 
 
-samplecand(rng, τ::StaticTrajectory{EndPointTS}, h, z) = step(τ.integrator, h, z, τ.n_steps)
+function samplecand(rng, τ::StaticTrajectory{EndPointTS}, h, z)
+    z′ = step(τ.integrator, h, z, τ.n_steps)
+    # Are we going to accept the `z′` via MH criteria?
+    is_accept, α = mh_accept_ratio(rng, energy(z), energy(z′))
+    return z′, is_accept, α
+end
 
 ### Multinomial sampling from trajecory
 
-randcat(rng::AbstractRNG, zs::AbstractVector{<:PhasePoint}, unnorm_ℓp::AbstractVector) = zs[randcat_logp(rng, unnorm_ℓp)]
+function randcat(rng::AbstractRNG, zs::AbstractVector{<:PhasePoint}, unnorm_ℓp::AbstractVector)
+    p = exp.(unnorm_ℓp .- logsumexp(unnorm_ℓp))
```

Comment:
Maybe separate this out into it's own function, say, `unnorm_logprob_to_prob`, and call `randcat(rng, zs, unnorm_logprob_to_prob(unnorm_ℓp))` instead? Basically like `randcat_logp` but separated into two functions.
Right now there are some impls of `randcat` that expects unnormalized log-probs while some expect probs :confused:

Reply:
I think what you proposed is nicer than the one I reverted, and I see what's your concern about `randcat` sometimes expecte unormed. log. prob. but sometimes expect normed. prob. 

The reason I reverted the previous `randcat_logp` is that this function is only called once in thie `randcat` and not reused at all. This will be the same issue for `unnorm_logprob_to_prob`, and I would prefer we don't make extract this into a seperate function until this is really used more than one time.

Maybe a better way to handle all of this here is actually renaming this `randcat` to `randcat_weight`. Then there will be no ambiguity: `randcat` alwasy expected normalised probability, and there will not function that is only used once.

Reply:
`randcat_weight` is mysterious I think; consider renaming `randcat` to `rand_categorical` if possible. I hesitated a bit to raise this issue in my previous review because this is not related to this PR, and can be fixed in a separate effort.